cmd/helm/init.go

```diff
@@ -92,8 +92,8 @@ func (i *initCmd) run() error {
 	return nil
 }
 
-func requireHome() error {
-	dirs := []string{homePath(), repositoryDirectory(), cacheDirectory(), localRepoDirectory()}
+func requireHome(home helmpath.Home) error {
```

Comment:
I'm opposed to turning a read-only function (search) into a read/write function. I feel like that could have unpleasant and hard to debug side-effects if someone's HELM_HOME is mis-set. But `ensureHome` seems way overblown for the single use case. Maybe I'll just add a quick os.Stat on the search function instead.